### DISCUSSION TITLE
Multi-author site set up

### DISCUSSION THREAD

**Discussion by rshmhrj** [jpanther/congo#138] [Q&A]
Hi all,
How would you go about using congo for a multi-author site?

Has anyone tried following the instructions [at CodingConcepts](https://codingnconcepts.com/hugo/multiple-authors-hugo/), [Netlify](https://www.netlify.com/blog/2018/07/24/hugo-tips-how-to-create-author-pages/), or [PolyGlotDeveloper](https://www.thepolyglotdeveloper.com/2018/04/manage-multiple-authors-static-blog-created-hugo/) with any luck?

It basically seems like adding an `author` front matter and reading the author details from a central `.json` file.

**Comment by jpanther** [comment]
Hey @rshmhrj, this is something that I have thought about implementing but you're the first person to ask for it.

Out of interest how many authors do you think you'd need for your site? I wonder if having a JSON file for author information is going to be too confusing for most Hugo users to understand instead of just storing it in the site config files.

**Comment by kiil** [comment]
I would also like this feature.

In my case there are four authors writing for the blog.

**Reply by kiil** [reply]
It seems Site.author and authors are deprecated in favor of using taxonomy for authors: https://github.com/gohugoio/hugo/issues/10551#issuecomment-1358572063

**Comment by wolfspyre** [comment]
I came up with a POC of this for a forked project, but I think it may be more beneficial to refactor for congo instead.

Will have a POC for this with Congo shortly to discuss implementation. 

Is this thread still the best place to discuss? or should a new thread be started for it?

**Reply by jpanther** [reply]
Here or a new thread is fine. Whichever you prefer!

**Comment by wolfspyre** [comment]
my general idea/plan is to adjust the singular author construct to something like 
 ‘manager’ ‘owner’ ‘marshal’ ‘automaton’ ‘minion’ ‘daemon’ … some pseudo abstract non-distinct entity… as the ‘voice of the site as a whole’ … think ‘announcer/town crier/majordomo’

and then have a taxonomy for  ‘contributors’ or something along those lines.

each contributor entity would be able to have adistinct value attribute for their title and i18n values…

I don’t think it would be outside of the realms of possibility to later incorporate a configurable list of roles that a contributor entity might have for different content; IE author for one post/series/article, editor for another, artist for another series or something… but i don’t want to overcomplicate the initial implementation.


are there any initial thoughts or concerns around the concept as is?

I have a functional implementation of this already; so it’s just a matter of restructuring and rewriting to fit elegantly. Shouldn’t take long…
(famous last words)

but I wanted to see if there were any specific needs peeps have already identified that I can try to ensure are at least supportable later if not supportable initially

**Reply by rshmhrj** [reply]
@wolfspyre, I wanted to create a "feedback" section in my blog with comments and quotes from previous managers and co-workers.  They would not be users, admins, authors, or contributors to the blog, but I wanted to attribute that particular "post" to them, and have their name / thumbnail image display on the page.

In Hugo terminology, I thought it would be multi-author, but that wasn't my specific use case.

**Reply by wolfspyre** [reply]
yeah… that’s outside the scope of multi-author IMO… … the intended purpose here would be to facilitate a piece of content having more than one contributing indivual from a cohort of more permanent / consistent contributors to a site construct…

there are several ways to accomplish what you’re describing.. each has its own bag of nuanced limitations and requirements… do you want to open a different q&a around commenters / submissions / reactions?

**Comment by kshpdr** [comment]
I also needed this feature so I ended up implementing a workaround for now. It's not perfect but it works so if anyone needs it urgently, you can check out [this commit](https://github.com/kshpdr/tuberlinru/commit/77856db8a2c028ad1887cf712359ba246864a07a) for [my website](https://tuberlin.ru) where you can find working examples.

Basically, after modifying html files you would need to create inside a `data/` folder an `authors/` folder and create for each author a corresponding toml file. After this you can directly specify an author inside each post like `authors: ["denis"]`

**Reply by wolfspyre** [reply]
sweet. My implementation is coming along... you can see where I'm at on the 
[contrib branch](https://gitlab.wolfspyre.io/wolfspyrelabs/hugo-kisso/wplshortcodes/-/tree/contrib?ref_type=heads)

Pretty close to a working contributor profile now,  just getting the article/articletype enumeration working. 

per-page attribution **shouldn't** be that hard to facilitate shortly thereafter. 
![contribList](https://github.com/jpanther/congo/assets/1680659/23054e16-468e-4716-b583-5f96425a67df)

**Comment by ItsMly** [comment]
This would be a great feature for congo - in my case I would need two authors

**Comment by diktamxx** [comment]
Is this feature available?

**Reply by wolfspyre** [reply]
I'm using it on my sites 
- https://hoof-paw.com/contributors  
- https://wolfspyre.com 

but I don't think as-is it's drop-in usable ... I've still got a fair bit of .... massaging to do to get things clean.... 

 if you wanna look at it, my code's here ...  https://gitlab.wolfspyre.io/wolfspyrelabs/hugo-kisso/wplshortcodes/-/tree/contrib

**Comment by vinceyoumans** [comment]
Thumbs up.   I would want multiple authos with an authors list page that drills down to their contribution content.
